server/ethereum/ethereum.test.js

```diff
@@ -45,12 +66,50 @@ beforeEach(async () => {
 
   pins = await node.pin.ls()
   if (pins.length > 0) throw new Error("Pins weren't removed properly.")
+
+  smartContracts.clear()
 })
 
 afterAll(() => {
   web3.currentProvider.connection.close()
 })
 
+test(`emitting listen event from listener, then emittting pin event
+from pinning contract (without registering pinner) pins file`, async done => {
+  const testKey = web3.utils.fromAscii('testKey')
+  const dag = { testKey: 'testVal' }
+  const hash = await node.dag.put(dag)
+
+  const emitPinEventAndCheck = () => {
+    contract.methods
+      .registerData(testKey, hash.toBaseEncodedString())
+      .send({ from: accounts[0] }, () => {
+        setTimeout(async () => {
+          const pins = await node.pin.ls()
+          const match = pins.find(item => {
+            return item.hash === hash.toBaseEncodedString()
+          })
+          expect(match).toBeDefined()
+          done()
+        }, 2000)
+      })
+  }
+
+  await listenerContract.methods
+    .unsubscribeContract(demoSmartContractJson1.address)
+    .send({ from: accounts[0] }, () => {
+      setTimeout(() => {}, 1000)
```

Comment:
Another option is just to make a utils function called `sleep` that we can use around the tests in all aspects, since we seem to be doing this a lot:

```js
const sleep = () => setTimeout(() => Promise.resolve(), 2000)
```

Reply:
oops just copied other tests, wasn't sure what it was doing